### DISCUSSION TITLE
Getting `Could not find a relationship between 'workout' and 'workout_exercise' in the schema cache` error on prod but not on local

### DISCUSSION THREAD

**Discussion by wcardosodev** [supabase/supabase#13026] [Questions] [answered]
I'm getting the above error in our prod build but not in our local build. My select query looks like this and works as expected locally:
```
  const { data } = await supabase
    .from('workout')
    .select(WORKOUT_SELECT)
    .eq('workout_id', workoutId);
```
Where `WORKOUT_SELECT` is:
```
  workout_id,
  name,
  date,
  time,
  is_completed,
  workout_exercise (
    workout_exercise_id,
    workout_id,
    exercise_id,
    sets,
    reps,
    weight,
    duration,
    order,
      exercise (
        exercise_id,
        name,
        description,
        exercise_set_type
      )
  )
```
This works locally but prod gives me the error. I've checked and ran `npx supabase db push` to which I get a message `Linked project is up to date` so it's not any missing migrations that haven't been run yet. My link on the tables is a foreign key on `workout_exercise` -> `workout` by `workout_id`.

**Comment by GaryAustin1** [comment, marked as answer]
Even though you say you have run migrations you might double check RLS policies are met on workout_exercise.

You could also run `NOTIFY pgrst, 'reload schema'` in the SQL editor, but that rarely is the issue.

**Reply by wcardosodev** [reply]
This is the RLS policy for select on the `workout_exercise` table:
```
(auth.uid() IN ( SELECT workout.user_id
   FROM workout
  WHERE (workout.workout_id = workout_exercise.workout_id)))
```
I'm trying to select workout_exercise rows where the workout_id column is linked to a workout which has a user_id column of the users user id.

I tried running `NOTIFY pgrst, 'reload schema'` just incase but didn't do anything afaik, still getting the same error

**Reply by GaryAustin1** [reply]
I was more asking if you checked that your RLS was enabled and the same between prod/local.  Not looking at whether it works or not as that should be the same in either case.
Double check your foreign key column shows up as an fk constraint (table UI is a quick way).

Also you could try restarting the database from dashboard general settings.  (once again a long shot).

**Reply by wcardosodev** [reply]
Ah I see, I thought that would've been the case but wanted to check. Yep in prod my foreign key from workout_exercise -> workout by workout_id is there as expected:
![image](https://user-images.githubusercontent.com/17492466/225369150-20c5cbab-96e6-4afd-aba0-1090615ff4c2.png)

Tried the restart aswell incase but also didn't solve it :(

**Reply by wcardosodev** [reply]
That image probably isn't the clearest, but the table it's on is the `workout_exercise` table and I clicked the link for the `workout_id` foreign key to show that it's linked to `workout.workout_id`

**Reply by GaryAustin1** [reply]
Your RLS does depend on auth.uid().  Any chance you are not signed in properly in one case and not the other?  Not sure if you are also running the client code differently.

**Reply by wcardosodev** [reply]
Just tried, I checked the login id and stuff on prod and it's got the correct email and looks (afaict) like it's logged in correctly, getting similar values from console logging `session.user` in both the prod / local versions.

**Reply by GaryAustin1** [reply]
Running out of ideas...  Can you temporarily turn off RLS on the two tables in PRODUCTION (not good idea if using it live)?  Just to rule out user id and RLS issue.   Then you are down to foreign key or corruption of some sort I guess.

Edit: And don't laugh... please double check your URL and key are to the instance you are updating...  Has happened before with multiple instances to be testing new changes on another "copy" of the instance with same tables.

**Reply by wcardosodev** [reply]
...It was that the dabase url and key were for an old copy of our DB that had a similar structure for some tables, so other pages of the app worked fine as they didnt change but the workout -> workout_exercise structure had changed so that's why that blew up...

Thank you so much for the help, wouldn't have realised that one as all the other pages worked as expected
